chutney/ui/src/app/modules/campaign/components/execution/detail/campaign-execution.component.ts

```diff
@@ -1,180 +1,303 @@
+            }
+        }
+        return this.UNSUPPORTED;
+    }
+
+    jiraLinkFrom(chutneyId: string) {
+        const foundScenario = this.jiraScenarios.find(s => s.chutneyId === chutneyId);
+        if (foundScenario) {
+            return this.jiraUrl + '/browse/' + foundScenario.id;
+        } else {
+            return null;
+        }
+    }
+
+    toQueryParams(scenarioExecutionReportOutline: ScenarioExecutionReportOutline): Params {
+        let execId = scenarioExecutionReportOutline.executionId !== -1 ? scenarioExecutionReportOutline.executionId : 'last';
+        return {
+            active: execId,
+            open: execId,
+        }
+    }
+
+    replay() {
+        this.campaignService.replayFailedScenario(this.report.report.executionId).subscribe({
+            error: (error) => this.eventManagerService.broadcast({ name: 'error', msg: error.error })
+        });
+
+        timer(1000).pipe(
+            switchMap(() => of(this.eventManagerService.broadcast({ name: 'replay', executionId: this.report.report.executionId })))
+        ).subscribe();
+    }
+
+    stop() {
+        this.campaignService.stopExecution(this.campaignId, this.report.report.executionId).subscribe({
+            error: (error) => this.eventManagerService.broadcast({ name: 'error', msg: error.error })
+        });
+    }
+
+    statusClass(scenarioReportOutline: ScenarioExecutionReportOutline): string {
+        if (scenarioReportOutline.status === ExecutionStatus.SUCCESS) {
+            return 'fa-solid fa-circle-check text-info';
+        }
+        if (scenarioReportOutline.status === ExecutionStatus.FAILURE) {
+            return 'fa-solid fa-circle-xmark text-danger';
+        }
+        if (scenarioReportOutline.status === ExecutionStatus.RUNNING || scenarioReportOutline.status === ExecutionStatus.PAUSED) {
+            return 'fa-solid fa-spinner fa-pulse text-warning';
+        }
+        if (scenarioReportOutline.status === ExecutionStatus.STOPPED) {
+            return 'fa-solid fa-circle-stop text-warning';
+        }
+        if (scenarioReportOutline.status === ExecutionStatus.NOT_EXECUTED) {
+            return 'fa-regular fa-circle text-warning';
+        }
+        return null;
+    }
+
+    sortBy(property) {
+        if (this.orderBy === property) {
+            this.reverseOrder = !this.reverseOrder;
+        }
+        this.orderBy = property;
+
+        return sortByAndOrder(
+            this.report.report.scenarioExecutionReports,
+            (i) => i[property] == null ? '' : i[property],
+            this.reverseOrder
+        );
+    }
+
+    exportReport() {
+        this.campaignService.findExecution(this.report.report.executionId)
+            .subscribe({
+                next: (report: CampaignExecutionFullReport) => {
+                    const url = location.origin + '/#' + this.router.serializeUrl(this.router.createUrlTree(['/scenario', ':id', 'executions']));
+                    const pdf = new jsPDF('l', 'mm', 'a4');

```

Comment:
Maybe could we extract this pdf creation to a utility function which takes a `CampaignExecutionFullReport` and returns a `jsPDF` type ?